contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -596,6 +596,16 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         return _checkIsAllowedRelayer(delivererAddress, allowedRelayers);
     }
 
+    /**
+     * @dev Sets the value of `blockchainID` to the value determined by the warp messenger precompile.
+     */
+    function initializeBlockchainID() public {
+        if (blockchainID != bytes32(0)) {
+            revert AlreadyInitialized();
+        }
+        blockchainID = WARP_MESSENGER.getBlockchainID();
```

Comment:
Since this is no longer set on deployment, we might want to emit an event on successful initialization.